### DISCUSSION TITLE
Blob files with react query

### DISCUSSION THREAD

**Discussion by tcvetkoska** [TanStack/query#3105] [Q&A]
Hello, can we download blob files with react-query?
Also, is there some mechanism that does the promise.all feature?

**Comment by TkDodo** [comment]
you can do whatever you want in your `queryFn`, as long as it returns a Promise. You can e.g. trigger multiple requests and return one Promise with `Promise.all`

**Reply by tcvetkoska** [reply]
Thanks, so there should not be any issues with downloading a big files with react-query?  Because of the caching mechanism I am afraid that it can cause some lagging and issues

**Reply by TkDodo** [reply]
what is your use-case please? Why would you want to cache a downloaded file?

**Reply by tcvetkoska** [reply]
We are starting to implement the react-query in our project and I actually want to be sure that I can use react-query, in any case, no matter the file type and to not worry about some edge cases that could happen. Also, I would not like to cache the file that I am downloading, maybe I missed somewhere in the documentation is there a possibility to not cache the requests with react-query?

**Reply by TkDodo** [reply]
> Also, I would not like to cache the file that I am downloading, maybe I missed somewhere in the documentation is there a possibility to not cache the requests with react-query?

no, there is no such possibility. If downloading a file is a one-off, possibly happening imperatively, I would just use a mutation for that.

**Reply by tcvetkoska** [reply]
thanks for the answers
